dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -14,8 +14,20 @@ WINE_DEFAULT_DEBUG_CHANNEL (shell);
 
 static HRESULT SHELL32_GetCLSIDForDirectory(LPCWSTR pwszDir, LPCWSTR KeyName, CLSID* pclsidFolder);
 
+static LPCWSTR GetItemFileName(PCUITEMID_CHILD pidl, LPWSTR Buf, UINT cchMax)
```

Comment:
Did you make it a separate function because you think you could use it elsewhere in the future?

Reply:
Yes, all places where it just goes directly for the wide string inside the PIDL are broken by design.